ras_backstage/controllers/controller.py

```diff
@@ -24,23 +30,95 @@ def build_url(service_name, config, proxy_path):
 
 
 @translate_exceptions
-def get_info():
+def get_info(config):
     info = {
-        "name": current_app.config['NAME'],
-        "version": current_app.config['VERSION'],
+        "name": config['NAME'],
+        "version": config['VERSION'],
+    }
+    info.update(config.get('METADATA', {}))
+
+    if config.feature.report_dependencies:
+        info["dependencies"] = [{'name': name} for name in config.dependency.keys()]
+
+    return info
+
+
+@translate_exceptions
+def sign_in(config, username, password):
+    oauth_svc = config.dependency['oauth2-service']
+    client_id = oauth_svc['client_id']
+    client_secret = oauth_svc['client_secret']
+
+    oauth_payload = {
+        'grant_type': 'password',
+        'username': username,
+        'password': password
     }
-    info.update(current_app.config.get('METADATA', {}))
 
-    if current_app.config.feature.report_dependencies:
-        info["dependencies"] = [{'name': name} for name in current_app.config.dependency.keys()]
+    oauth_url = '{}://{}:{}/api/v1/tokens/'.format(oauth_svc['scheme'], oauth_svc['host'], oauth_svc['port'])
+    payload = '&'.join(["{}={}".format(k, v) for k, v in oauth_payload.items()])
+    response = requests.post(oauth_url,
+                             auth=(client_id, client_secret),
+                             data=payload,
+                             headers={'Content-Type': "application/x-www-form-urlencoded"})
+    response.raise_for_status()
+    response_data = response.json()
+
```

Comment:
Think if you catch up with Shane on Monday and add some type of error handling for too many wrong sign in, locked account and account not active. Could just copy the code from ras_frontstage? No idea how this will work with the angular app. I'll talk to you on Monday.

Reply:
Requirements are unclear around this at the moment, so it's a bit hard to say what kinds of error we need to differentiate. For example, the user has to separately sign-in to R-Ops before even getting to our login, which I doubt is the final design for the system. Thus I'd prefer to avoid putting much effort in without the requirements being made explicit.